src/nvim/window.c

```diff
@@ -5158,6 +5160,9 @@ static void last_status_rec(frame_T *fr, int statusline)
  */
 int tabline_height(void)
 {
+  if (tabline_external) {
```

Comment:
Can this query the UI structure instead of storing extra state in window.c? E.g. `ui_is_widget_external(kUITabline)`, where the parameter is a enum like:

enum {
kUITabline,
kUIStatusline,
kUICmdline,
...
}